### PR TITLE
Daily code challenges

### LINKED ISSUE #74: Admin page for proggekonkurranse

### PR DESCRIPTION
**Pull request by danielyanghansen**
Resolves #74 #58 

Working admin page under /admin/codeCompetition

Still missing PUT implementation for competitions

### CODE REVIEW

**Comment by Arashfa0301** [comment]
There exists a weird bug when you open the first competition and want to close it again. It closes and opens again. Could you look into it?

**Review by eikhr — DISMISSED** [review]
Looks like you might have installed dependencies using npm instead of yarn, this has created a `package-lock.json` file instead of updating `yarn.lock`. Just delete `package-lock.json` and run `yarn` to update `yarn.lock`

**Comment by danielyanghansen** [comment]
> There exists a weird bug when you open the first competition and want to close it again. It closes and opens again. Could you look into it?

It looks like the bug comes from the competition ordering, as it is sorted by date and updated accordingly. The solution is probably not very complicated, but just having one competition per day might be a simple workaround

File: components/DailyCompetition.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      <Box></Box>
+      <ReactMarkdown remarkPlugins={[remarkGfm]}>
+        {competition.description}
+      </ReactMarkdown>
+    </Card>
+  );
+};
+
+export default DailyCompetitionElement;
+/**/
```
Comment (eikhr) [inline components/DailyCompetition.tsx:48]:
```suggestion

```

File: components/DailyCompetition.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  const deleteElement = async () => {
+    await fetch(`/api/dailyCompetitions/${competition._id as string}`, {
+      method: "DELETE",
+      headers: {
+        password: adminPassword ?? "",
+        "Content-Type": "application/json",
+      },
+    });
+  };
+  //TODO: Style Box:
```
Comment (eikhr) [inline components/DailyCompetition.tsx:22]:
```suggestion
```
Done?

File: models/codeCompetitionSchema.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  description: string;
+  date: string;
+  index: number;
+}
+
+const schema = new Schema<DailyCompetition>({
+  title: { type: String, default: "" },
+  description: { type: String, default: "" },
+  date: { type: String, default: "" },
+  index: { type: Number, default: 0 },
```
Comment (eikhr) [inline models/codeCompetitionSchema.ts:16]: What is the index used for? They are currently sorted by date

File: pages/api/dailyCompetitions/[id].ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const { method, headers, query } = req;
+
+  await mongoose.connect(url);
+
+  switch (
+    method //TODO: Case when there is nothing to return/delete
```
Comment (eikhr) [inline pages/api/dailyCompetitions/[id].ts:15]: don't check in todos, please

**Review by eikhr — APPROVED** [review]
Looks good :) have commented on some small issues

**Pull request merged**